src/Microsoft.HttpRepl/ApiConnection.cs

```diff
@@ -81,31 +80,51 @@ public async Task FindSwaggerDoc(HttpClient client, IEnumerable<string> swaggerS
             }
         }
 
-        public async Task<string> GetSwaggerDocAsync(HttpClient client, Uri uri, CancellationToken cancellationToken)
+        private async Task<(bool Success, string Document)> GetSwaggerDocAsync(HttpClient client, Uri uri, VerbosityLogger logger, CancellationToken cancellationToken)
         {
-            var resp = await client.GetAsync(uri, cancellationToken).ConfigureAwait(false);
-            if (cancellationToken.IsCancellationRequested)
+            try
             {
-                return null;
-            }
-
-            resp.EnsureSuccessStatusCode();
-            string responseString = await resp.Content.ReadAsStringAsync().ConfigureAwait(false);
+                logger.WriteVerbose(Resources.Strings.ApiConnection_Logging_Checking, uri);
+                var response = await client.GetAsync(uri, cancellationToken).ConfigureAwait(false);
+                if (cancellationToken.IsCancellationRequested)
+                {
+                    logger.WriteLineVerbose(Resources.Strings.ApiConnection_Logging_Cancelled);
+                    return (false, null);
+                }
 
-            return responseString;
-        }
+                if (response.IsSuccessStatusCode)
+                {
+                    logger.WriteLineVerbose(Resources.Strings.ApiConnection_Logging_Found);
+                    string responseString = await response.Content.ReadAsStringAsync().ConfigureAwait(false);
 
-        public async Task<(bool Success, string Document)> TryGetSwaggerDocAsync(HttpClient client, Uri uri, CancellationToken cancellationToken)
-        {
-            try
-            {
-                string document = await GetSwaggerDocAsync(client, uri, cancellationToken);
-                return (true, document);
+                    logger.WriteVerbose(Resources.Strings.ApiConnection_Logging_Parsing);
+                    ApiDefinitionReader reader = new ApiDefinitionReader();
+                    if (reader.CanHandle(responseString))
+                    {
+                        logger.WriteLineVerbose(Resources.Strings.ApiConnection_Logging_Successful);
+                        return (true, responseString);
+                    }
+                    else
+                    {
+                        logger.WriteLineVerbose(Resources.Strings.ApiConnection_Logging_Failed);
+                        return (false, null);
+                    }
+                }
+                else
+                {
+                    logger.WriteLineVerbose(response.StatusCode.ToString());
+                    return (false, null);
+                }
             }
-            catch
+            catch (Exception e)
             {
+                logger.WriteLineVerbose(e.Message);
```

Comment:
What's the error experience when not using verbose?  Should we always show all error details in non-verbose?

Reply:
There's two scenarios that use this code. The first, when you specify an OpenAPI Description directly. The second, when we're using the list of possible OpenAPI Description paths to check a bunch of them to find the first one that works. 

Currently (prior to this PR) the end result is the same in both scenarios - either it found a swagger file, or it didn't and we say where we found it if we did. WHY we didn't find it (or didn't find several) is not shown. 

For the former case, I could see an argument for always showing the why. For the latter, we wouldn't want the user to see a bunch of failures (for however many times it fails before it succeeds) - that's where the verbose use case came from. 

So, at least currently, its a little odd because the exception isn't necessarily unexpected depending on the codepath. Maybe I can enable verbose for the directly-specified scenario automatically, with the need for the option?

Reply:
I went ahead and enabled verbose for ApiConnection when specifying the swagger file directly, which I think is helpful and in the spirit of the ask for the related issue anyway.